modules/oe_webtools_etrans/src/Plugin/Block/UnifiedEtransBlock.php

```diff
@@ -0,0 +1,309 @@
+          'target' => '_blank',
+        ],
+      ]),
+      '#title' => $this->t('Important information about machine translation', [], ['langcode' => $translation_to_language->getId()]),
+    ];
+    $build['translation_request'] = [
+      '#theme' => 'block__unified_etrans_request',
+      '#message' => $message,
+      '#translate_link' => $translate_link,
+      '#disclaimer_link' => $disclaimer_link,
+    ];
+
+    return $build;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration(): array {
+    return self::DEFAULT_CONFIGURATION + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state): array {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    $form['delay'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Delay'),
+      '#min' => 0,
+      '#description' => $this->t('The time in milliseconds to delay rendering the translation. Use this on dynamic pages if the HTML element that contains the translation is not immediately available.'),
+      '#default_value' => $this->configuration['delay'],
+    ];
+
+    $form['include'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Include'),
+      '#description' => $this->t('A list of CSS selectors indicating the page elements to be translated, one selector per line. If omitted the entire page will be translated.'),
+      '#default_value' => (string) $this->configuration['include'],
+    ];
+
+    $form['exclude'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Exclude'),
+      '#description' => $this->t('A list of CSS selectors indicating page elements to be excluded from the translation even if they are inside an "include" element. One selector per line.'),
+      '#default_value' => (string) $this->configuration['exclude'],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state): void {
+    parent::submitConfigurationForm($form, $form_state);
+
+    foreach (array_keys(self::DEFAULT_CONFIGURATION) as $key) {
+      $this->configuration[$key] = $form_state->getValue($key);
+    }
+
+    $this->configuration['delay'] = (int) $form_state->getValue('delay');
+  }
+
+  /**
+   * Get route entity langcode (translation or default).
+   *
+   * @param bool $default
+   *   Whether to fallback to node default site language on node route.
+   *
+   * @return string
+   *   Requested langcode, default one or empty string.
+   */
+  protected function getRouteEntityLangcode(bool $default = FALSE): string {
+    $route = $this->routeMatch->getRouteObject();
+    $requested_langcode = $this->languageManager->getCurrentLanguage()->getId();
+    if ($route->hasRequirement('node')) {
+      /** @var \Drupal\node\NodeInterface $entity */
+      $entity = $this->routeMatch->getParameter('node');
```

Comment:
Are we sure that entity is an instance of NodeInterface at this point?

Reply:
Except if someone implements {node} parameter as expecting a taxonomy term in their route, we should expect a NodeInterface yes. Anyway, this code is only informative and does not impact its computing. Now, most of the function called later on are actually depending on TranslatableEntityInterface, so renaming the variable $node and keeping NodeInterface, or keeping $entity and indicating TranslatableEntityInterface ... At the end, it'll all depend on what other modules send in their routes with the parameter {node}, which we can't predict.